test/integration/targets/lookup_vars/tasks/main.yml

```diff
@@ -6,14 +6,25 @@
     that:
       - 'var_host == ansible_host'
 
+- name: Test that we can give multiple values to var and receive a list of values back
+  block:
+    - set_fact:
+        var_host_info: '{{ query("vars", "ansible_host", "ansible_connection") }}'
+
+    - assert:
+        that:
+          - 'var_host_info[0] == ansible_host'
+          - 'var_host_info[1] == ansible_connection'
+
 - name: Test that we can give a list of values to var and receive a list of values back
-  set_fact:
-    var_host_info: '{{ query("vars", "ansible_host", "ansible_connection") }}'
+  block:
+    - set_fact:
+        var_host_info: '{{ query("vars", ["ansible_host", "ansible_connection"]) }}'
 
-- assert:
-    that:
-      - 'var_host_info[0] == ansible_host'
-      - 'var_host_info[1] == ansible_connection'
+    - assert:
+        that:
+          - 'var_host_info[0] == ansible_host'
+          - 'var_host_info[1] == ansible_connection'
```

Comment:
This diff would be a lot easier to read if you had stuck with the existing formatting instead of moving things inside blocks which have no real utility (and result in neither of the tasks having names.)

Reply:
@flowerysong thanks for the feedback. My reasoning was based on the fact the there are 2 blocks doing more or less the same thing (except the way the lookup plugin is being called). To prevent confusion and clarify scope I thought: let's group this in a block, but I'll keep it in mind and prevent doing that next time.